### DISCUSSION TITLE
Airtable Read Query - Only returning 10 records if NumRecords set to default

### DISCUSSION THREAD

**Discussion by ghost** [Budibase/budibase#7312] [Help]
Hi all - I seem to be having some problems with my Airtable read query. 

I have an Airtable that is set up and working. In have the fields set up for the query as follows:

![image](https://user-images.githubusercontent.com/64409787/184926768-16ef7101-8983-48a6-8695-3196bd9c6acc.png)

When I try to display this data in the designer, only 10 records are being displayed. Only when I set the NumRecords field to e.g. 100 are more records being shown. I had thought by leaving the NumRecords field blank it would return all records by default. 

Am I being dumb or is there an underlying problem here?

Thanks

**Comment by noseshimself** [comment]
That's pretty normal; Airtable is delivering data in "pages" and the default page size is 10. A lot of Django applications is doing it that way (I've wasted weeks of my life writing a python interface to Mayan DMS) and annoying me incredibly...

**Reply by ghost** [reply]
Hi @noseshimself 

Are you sure the default page size is 10? This is straight from the API docs
![image](https://user-images.githubusercontent.com/64409787/185071622-8183777b-95b5-4db5-8383-20f0dffca298.png)

Additionally, in the budibase docs they say the following:

> The `NumRecords` field is an optional field, by default all data will be returned, but if you want to limit the results you could enter a number here.

I'm just a bit confused by what I'm being told should happen and what is actually happening.

Thanks

**Reply by noseshimself** [reply]
If I'm using the default settings I'm getting 10 per "page". And as it is closed source it's more than trivial work to peek inside. I stopped using Airtable and went to nocodb for anything airtaibly I had to get done and hope to replace that by budibase.

**Comment by melohagan** [comment]
Hey @adamjdavies 

Looking at the codebase, the default is hardcoded to be 10:
![Screenshot 2022-08-18 at 09 49 00](https://user-images.githubusercontent.com/101575380/185353011-6c0e820c-86bf-4619-9db0-d09068d2e357.png)
